balanced-parentheses/balanced-parentheses-helper.js

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable require-jsdoc */
+/* eslint-disable camelcase */
+
+export function is_balanced(expression) {
+  try {
+    console.log("EXPRESSION TO CHECK: ", expression);
+    if (!expression || expression.length == 0 ) return false;
+
+    const stack = [];
+    const Paranthesis_map = {
+      "(": ")",
+      "{": "}",
+      "[": "]",
+    };
+    const close_parantheses = [")", "}", "]"];
+
+    for (let i = 0; i < expression.length; i++) {
+      if (expression[i] in Paranthesis_map) {
+        stack.push(expression[i]);
+      } else if (
+        close_parantheses.includes(expression[i]) &&
+        expression[i] != Paranthesis_map[stack.pop()]
+      ) {
+        return false;
+      }
+    }
+
+    if (stack.length > 0) {
+      return false;
+    }
+
+    return true;
+  } catch (err) {
```

Comment:
Why did we use try catch here?

Reply:
I thought when we pop from empty stack it will give exception that's why put try, catch. But now I tried It gives undefined